### ISSUE TITLE
Unrecognized GEDCOM code (2.0.0. beta 4)

### ISSUE THREAD

**Issue by c-prompt** [fisharebest/webtrees#2589]
Import from ancestry.com displays _EMPLOY for Employment (_EMPLOY in .ged file). When I look at the HTML, I see:

`<span class="error" title="Unrecognized GEDCOM code">_EMPLOY</span>`

**Comment by c-prompt** [comment]
Actually, I also see this for _APID - see https://support.ancestry.com/s/question/0D51500001jn4sMCAQ/where-is-the-definition-of-the-apid-tag-in-an-exported-gedcom-file-from-a-family-tree

**Comment by fisharebest** [comment]
Tags beginning with `_` are custom tags.

GEDCOM standards say that genealogy applications should **ignore** all unrecognised custom tags.

For this reason, and the long-term security of your genealogy data, you should **avoid** custom tags as far as possible.  (There are a few widely used exceptions to this rule, such as `_NMR`, '_ASSO` and `_MARNM`.)

There's a configuration option in the family-tree preferences page "Show GEDCOM errors".
It controls whether webtrees ignores or displays these unrecognised tags.

So, this is all working as designed.

Ancestry.com have a habit (policy?) of creating non-standard GEDCOM files to make it difficult to migrate your data away from them.  For example, they could have used `1 OCCU`/`2 AGNC` to record employment.

Now, webtrees has (for historical reason) included basic support for a number of custom tags.  By support, we simply have a translation for the label.  No meaning is ascribed to these tags.

My preference would be to remove this support - although removing functionality is never popular.
Another option might be to make it possible to support custom tags through the module system.

`_APID` is presumably "Ancestry.com permanent identifier"

Can you post examples of GEDCOM data for both these tags, so that I can see what subtags (if any) they use?

**Comment by c-prompt** [comment]
Makes complete sense and thanks - I wasn't aware of that.

Here's what I see in the .ged file:

```
1 _EMPLOY farmer
2 OBJE
3 FILE http://trees.ancestry.com/rd?f=document&guid=removed&tid=removed&pid=11
3 FORM pdf
3 TITL  A letter describing his farming
```

```
0 @P12@ INDI 
1 BIRT 
2 DATE 18 May 1921
2 PLAC France
2 SOUR @removed@
3 PAGE Year: 1910; Census Place: removed
3 _APID 1,1232::51512345
2 SOUR @removed@
3 PAGE The National Archives in St. Louis, Missouri; removed
4 CONC : 552
3 _APID 1,9876::40612345
2 SOUR @removed@
3 PAGE Year: 1912; Arrival: New York, New York; Microfilm Serial: removed
3 _APID 1,12345::20112345
2 SOUR @removed@
3 PAGE Year: 1912; Arrival: New York, New York; Microfilm Serial: also removed
3 _APID 1,4343::201123456
```

**Comment by fisharebest** [comment]
This fix has been merged into the main branch, and will be included in 2.1.0.
